bin/pycbc_compress_bank

```diff
@@ -146,23 +178,40 @@ for ii in range(imin, imax):
     htilde = bank[ii-imin]
     tmplt = bank.table[ii-imin]
     kmax = htilde.end_idx
-    
-    # get the compressed sample points
-    if args.compression_algorithm == 'mchirp':
-        sample_points = compress.mchirp_compression(tmplt.mass1, tmplt.mass2,
-            fmin, (kmax-1)*df, min_seglen=args.t_pad, df_multiple=df).astype(
-            real_same_precision_as(htilde))
-    elif args.compression_algorithm == 'spa':
-        sample_points = compress.spa_compression(htilde, fmin, (kmax-1)*df,
-            min_seglen=args.t_pad).astype(real_same_precision_as(htilde))
+    tmplt_approximant = htilde.approximant
+    if num_compression_algorithms == 2:
+        if tmplt_approximant == approximant1:
+            compression_type = compression_type1
+        if tmplt_approximant == approximant2:
+            compression_type = compression_type2
+    # Call the appropriate function in compress.py to get the compressed
+    # amplitude and phase frequency points and decompress them in
+    # frequency space :
+
+    # If template is SEOBNRv2_ROM perform linear_compression or
+    # partial_ROM_compression as per user input
+    if compression_type == 'mchirp':
+        sample_points = compress.mchirp_compression(
+            tmplt.mass1, tmplt.mass2, fmin, (kmax-1)*df,
+            min_seglen=args.t_pad,
+            df_multiple=df).astype(real_same_precision_as(htilde))
+        hcompressed = compress.compress_waveform(
+            htilde, sample_points, args.tolerance, args.interpolation,
+            decomp_scratch=decomp_scratch)
+    elif compression_type == 'spa':
+        sample_points = compress.spa_compression(
+            htilde, fmin, (kmax-1)*df, min_seglen=
+            args.t_pad).astype(real_same_precision_as(htilde))
+        hcompressed = compress.compress_waveform(
+            htilde, sample_points, args.tolerance, args.interpolation,
+            decomp_scratch=decomp_scratch)
+    elif compression_type == 'partial_compress_rom':
+        hcompressed = compress.partial_rom_compression(
+            htilde, tmplt.mass1, tmplt.mass2, tmplt.spin1z, tmplt.spin2z,
+            df, fmin, (kmax-1)*df, args.interpolation)
```

Comment:
It might be good to move this decision making about what compress function to call as a function to the compress module (à la `get_fd_waveform`). Then, as you add compression algorithms in the future, you need only edit the compress module and not touch the executable. But I'm fine with leaving this here for now; that can be addressed in a future PR.